### DISCUSSION TITLE
Using chakra/material ui in nextjs

### DISCUSSION THREAD

**Discussion by NaviTheCoderboi** [vercel/next.js#56461] [App Router] [answered]
Guys I have a question that like mui/chakra ui is recommended for big apps , I want to create multiple themes in mui but this would need to setup a provider in layout.tsx which would make whole app client side and I won't be able to do server side rendering, is there any alternative to this or I shouldn't use materialui? Or I use any other framework than nextjs?

**Comment by Sinista121** [comment]
You can actually use ChakraUI in server components by making a client component, which exports the Chakra Lib, and then importing from this component file. 

For example, you can have a file named `Chakra.js` and write the following code :
```js
'use client';

export * from '@chakra-ui/react';
```

now in a server component, you can just do this:
```js
import Chakra from "./Chakra.js";
```

**Comment by michael-bouwer** [comment, marked as answer]
You can use a composition pattern mentioned here [https://nextjs.org/docs/app/building-your-application/rendering/composition-patterns#using-context-providers](here). This won't render your whole app on the client, but rather just the provider.

More info on this pattern can be found here [https://nextjs.org/docs/app/building-your-application/rendering/composition-patterns#supported-pattern-passing-server-components-to-client-components-as-props](here)
